WordPress/src/test/java/org/wordpress/android/ui/posts/editor/media/AddLocalMediaToPostUseCaseTest.kt

```diff
@@ -0,0 +1,286 @@
+        val copyMediaToAppStorageUseCase = createCopyMediaToAppStorageUseCase(
+                createCopyMediaResult(copyingSomeMediaFailed = true)
+        )
+        // Act
+        val result = createAddLocalMediaToPostUseCase(copyMediaToAppStorageUseCase = copyMediaToAppStorageUseCase)
+                .addNewMediaToEditorAsync(localUris, SITE_MODEL, FRESHLY_TAKEN, mock())
+        // Assert
+        assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `addNewMediaToEditorAsync returns false when creating media model fails`() = test {
+        // Arrange
+        val localUris = listOf<Uri>(mock(), mock())
+        val getMediaModelUseCase = createGetMediaModelUseCase(
+                createMediaModelResult(
+                        loadingSomeMediaFailed = true
+                )
+        )
+        // Act
+        val result = createAddLocalMediaToPostUseCase(getMediaModelUseCase = getMediaModelUseCase)
+                .addNewMediaToEditorAsync(localUris, SITE_MODEL, FRESHLY_TAKEN, mock())
+        // Assert
+        assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `Uris of copied files are passed to optimizeMediaUseCase`() = test {
+        // Arrange
+        val localUris = listOf<Uri>(mock(), mock())
+        val urisOfCopiedFiles = listOf<Uri>(mock())
+        val copyMediaToAppStorageUseCase = createCopyMediaToAppStorageUseCase(
+                createCopyMediaResult(uris = urisOfCopiedFiles)
+        )
+
+        val optimizeMediaUseCase = createOptimizeMediaUseCase()
+        // Act
+        createAddLocalMediaToPostUseCase(
+                copyMediaToAppStorageUseCase = copyMediaToAppStorageUseCase,
+                optimizeMediaUseCase = optimizeMediaUseCase
+        ).addNewMediaToEditorAsync(localUris, SITE_MODEL, FRESHLY_TAKEN, mock())
+        // Assert
+        verify(optimizeMediaUseCase).optimizeMediaIfSupportedAsync(
+                any(),
+                eq(FRESHLY_TAKEN),
+                eq(urisOfCopiedFiles)
+        )
+    }
+
+    @Test
+    fun `Uris of optimized files are passed to getMediaUseCase`() = test {
+        // Arrange
+        val localUris = listOf<Uri>(mock(), mock())
+        val urisOfOptimizedFiles = listOf<Uri>(mock())
+        val optimizeMediaUseCase = createOptimizeMediaUseCase(createOptimizeMediaResult(uris = urisOfOptimizedFiles))
+
+        val getMediaModelUseCase = createGetMediaModelUseCase()
+        // Act
+        createAddLocalMediaToPostUseCase(
+                optimizeMediaUseCase = optimizeMediaUseCase,
+                getMediaModelUseCase = getMediaModelUseCase
+        ).addNewMediaToEditorAsync(localUris, SITE_MODEL, FRESHLY_TAKEN, mock())
+        // Assert
+        verify(getMediaModelUseCase).createMediaModelFromUri(
+                eq(LOCAL_SITE_ID),
+                eq(urisOfOptimizedFiles)
+        )
+    }
+
+    @Test
+    fun `MediaModel upload status is set to QUEUED`() = test {
+        // Arrange
```

Comment:
are we doing this now? I went back and forth between `Given/When/Then` and not using anything and tbh either approach works for me 🤷‍♂

Reply:
We started using it in our team about 3 months ago and I kind of like it. It doesn't really matter what words `Given/When/Then vs Arrange/Act/Assert` we use, but I like the separation.